elementary/monitor/alerts/group_of_alerts.py

```diff
@@ -0,0 +1,331 @@
+
+    def set_owners(self, owners: list[str]):
+        self._components_to_attention_required[OwnersComponent] = ", ".join(owners)
+
+    def set_subscribers(self, subscribers: list[str]):
+        self._components_to_attention_required[SubsComponent] = ", ".join(subscribers)
+
+    def _sort_channel_destination(self, default_channel):
+        raise NotImplementedError
+
+    def _fill_components_to_alerts(self):
+        test_errors = []
+        test_warnings = []
+        test_failures = []
+        model_errors = []
+        for alert in self.alerts:
+            if isinstance(alert, ModelAlert):
+                model_errors.append(alert)
+            elif alert.status == "error":
+                test_errors.append(alert)
+            elif alert.status == "warn":
+                test_warnings.append(alert)
+            else:
+                test_failures.append(alert)
+        self._components_to_alerts: Dict[AlertGroupComponent, List[Alert]] = dict()
+        if model_errors:
+            self._components_to_alerts[ModelErrorComponent] = model_errors
+        if test_failures:
+            self._components_to_alerts[TestFailureComponent] = test_failures
+        if test_warnings:
+            self._components_to_alerts[TestWarningComponent] = test_warnings
+        if test_errors:
+            self._components_to_alerts[TestErrorComponent] = test_errors
+
+    def to_slack(self) -> SlackMessageSchema:
+        title_blocks = []  # title, [banner], number of passed or failed,
+        title_blocks.append(self._title_block())
+        banner_block = self._get_banner_block(self._env)
+        if banner_block:
+            title_blocks.append(banner_block)
+        title_blocks.append(self._number_of_failed_block())
+        self._message_builder._add_title_to_slack_alert(title_blocks=title_blocks)
+
+        # attention required : tags, owners, subscribers
+        self._message_builder._add_preview_to_slack_alert(
+            preview_blocks=self._attention_required_blocks()
+        )
+
+        details_blocks = []
+        for component, alerts_list in self._components_to_alerts.items():
+            details_blocks.append(
+                self._message_builder.create_text_section_block(
+                    f":{component.emoji_in_summary}: *{component.name_in_summary}*"
+                )
+            )
+            details_blocks.append(self._message_builder.create_divider_block())
+            if component == ModelErrorComponent:
+                blocks = self._get_model_error_blocks()
+                details_blocks.extend(blocks)
+            else:
+                text = self._tabulate_list_of_alerts(alerts_list)
+                details_blocks.append(
+                    self._message_builder.create_text_section_block(text)
+                )
+        self._message_builder._add_blocks_as_attachments(details_blocks)
+
+        return self._message_builder.get_slack_message()
+
+    def _title_block(self):
```

Comment:
I think that those methods should be part of the SlackMessageBuilder of group alerts.
If we want to do a separation between the data formatting and the platform the alert is sent to (Slack / MS Teams), we probably should move the specific Slack methods under the SlackMessageBuilder and leave the GroupOfAlerts class only with the to_slack or something 🤔

Reply:
After talking with @RoiTabach we understood that it is ok to have functions that returns content of alert as part of the class as long as they don't relay on a message builder, and move only the parts that does relay on it.